packages/babel-parser/test/helpers/run-fixture-tests.js

```diff
@@ -93,6 +93,13 @@ function runParseTest(parse, test, onlyCompareErrors) {
   }
 
   const actual = parseWithRecovery(parse, source, filename, options);
+
+  // When errorRecovery is true/always we can't trust the structure of the
+  // AST (other than the attached errors), so only compare errors.
+  onlyCompareErrors =
+    onlyCompareErrors ||
+    (actual.ast && actual.ast.errors && actual.ast.errors.length > 0);
```

Comment:
In most cases, with recoverable errors the AST will be correct. Can we use a check like this, to only skip comparing the AST if we had "unrecoverable" errors?
```suggestion
  // When errorRecovery is true/always we can't trust the structure of the
  // AST (other than the attached errors), so only compare errors.
  onlyCompareErrors =
    onlyCompareErrors ||
    (actual.ast && actual.ast.errors?.length > 0 && isEmptyResult(actual.ast));
  
  function isEmptyResult(ast) {
    if (ast.type === "SequenceExpression") return ast.expressions.length === 0;
    if (ast.type === "File") {
      const { program } = ast;
      return program.body.length === 0 && program.directives.length === 0 && !program.interpreter;
    }
    return false;
  }
```